frontend/app/src/components/home/ChatMessageMenu.svelte

```diff
@@ -260,13 +290,66 @@
             }
         });
     }
+
+    // Note: Manually selected reactions do not increment their fav counter!
+    function toggleReaction(reaction: string) {
+        if (canReact) {
+            const kind = client.containsReaction(createdUser.userId, reaction, msg.reactions)
+                ? "remove"
+                : "add";
+
+            client
+                .selectReaction(
+                    chatId,
+                    createdUser.userId,
+                    threadRootMessageIndex,
+                    msg.messageId,
+                    reaction,
+                    createdUser.username,
+                    createdUser.displayName,
+                    kind,
+                )
+                .then((success) => {
+                    if (success && kind === "add") {
+                        client.trackEvent("reacted_to_message");
+                    }
+                });
+        }
+    }
 </script>
 
-<div class="menu" class:rtl={$rtlStore}>
+<div class="menu" class:inert={inert} class:rtl={$rtlStore}>
+    {#await loadedReactions }
+        <div></div>
+    {:then [fst, snd, thrd]}
+        {#if !inert}
+            {#if "unicode" in fst}
+                <HoverIcon compact={true} onclick={() => toggleReaction(fst.unicode)}>
+                    <div class="quick-reaction">
+                        {fst.unicode}
+                    </div>
+                </HoverIcon>
+            {/if}
+            {#if "unicode" in snd}
+                <HoverIcon compact={true} onclick={() => toggleReaction(snd.unicode)}>
+                    <div class="quick-reaction">
+                        {snd.unicode}
+                    </div>
+                </HoverIcon>
+            {/if}
+            {#if "unicode" in thrd}
+                <HoverIcon compact={true} onclick={() => toggleReaction(thrd.unicode)}>
+                    <div class="quick-reaction">
+                        {thrd.unicode}
+                    </div>
+                </HoverIcon>
+            {/if}
+        {/if}
+    {/await}
     <MenuIcon bind:this={menuIcon} centered position={"right"} align={"end"}>
         <div class="menu-icon" slot="icon">
             <HoverIcon compact>
-                <ChevronDown size="1.6em" color={me ? "#fff" : "var(--icon-txt)"} />
+                <DotsVertical size="1.4em" color="#fff" />
```

Comment:
This colour will probably cause theme issues and I would suggest that we use `var(--menu-txt)` rather than `#fff`.